cmd/telemetrygen/go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/open-telemetry/opentelemetry-collector-contrib/cmd/telemetrygen
 
-go 1.22.0
+go 1.22.7
```

Comment:
This should be reverted, see https://github.com/grpc/grpc-go/commit/3db86e2e4b397d4139c0bc84be81ed9e70bd98ab